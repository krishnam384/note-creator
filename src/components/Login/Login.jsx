import React, { useRef, useState, useEffect } from "react";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import { Link } from "react-router-dom";
// import { useHistory } from "react-router-dom";
// import { useDispatch } from "react-redux";
// import { store } from "react-notifications-component";
// import axios from "axios";
// import { handleError } from "../../reducer/action";
import { emailValidation, passwordValidation } from "../../utils/validators";
import Error from "../ErrorPages/Error";
import "./Login.css";

export default function Login() {
  // const dispatch = useDispatch();
  // const history = useHistory();
  const emailFocus = useRef();

  // const token = localStorage.getItem("token");

  // const handleSubmit = (event) => {
  //   event.preventDefault();
  //   console.log("Submit button is Clicked..!!");

  //   setError(validate(user));
  //   console.log(token);
  //   console.log(user);
  //   const headers = {
  //     "content-type": "application/json",
  //     Authorization: token,
  //   };

  //   axios
  //     .post("http://localhost:8000/api/v1/auth/login", user, {
  //       headers: headers,
  //     })
  //     .then((response) => {
  //       if (response.status === 200) {
  //         console.log("User Logged in successfully");
  //         console.log(response);

  //         console.log(
  //           store.addNotification({
  //             title: "Successful..!!",
  //             message: "User Logged in Successfully..!!",
  //             type: "success",
  //             insert: "top",
  //             container: "top-right",
  //             animationIn: ["animate__animated", "animate__fadeIn"],
  //             animationOut: ["animate__animated", "animate__fadeOut"],
  //             dismiss: {
  //               duration: 2000,
  //               onScreen: true,
  //             },
  //           })
  //         );
  //         history.push("/home");
  //       }
  //     })
  //     .catch((e) => dispatch(handleError(e.message, props)));
  // };

  // useEffect(() => {
  //   emailFocus.current.focus();
  //   console.log(error);
  // }, [error]);

  const [user, setUser] = useState({ email: "", password: "" });
  const [isError, setIsError] = useState(false);
  const [formValidationErrors, setFormValidationErrors] = useState({
    emailError: "",
    passwordError: "",
  });

  let loginFormValidationState = false;
  let FormErrors = { emailError: "", passwordError: "" };

  const handleChange = (event) => {
    const { name, value } = event.target;

    setUser({ ...user, [name]: value });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    console.log(user);
    validateLoginForm(user);
    setFormValidationErrors({ ...FormErrors });
    if (!loginFormValidationState) {
      console.log("Perform Fetch Operation");
    }
  };

  const validateLoginForm = (user) => {
    FormErrors.emailError = emailValidation(user.email);
    FormErrors.passwordError = passwordValidation(user.password);
    console.log(FormErrors.emailError);
    if (FormErrors.emailError !== "" || FormErrors.passwordError !== "") {
      loginFormValidationState = true;
      setIsError(true);
    } else {
      loginFormValidationState = false;
      setIsError(false);
    }
  };

  useEffect(() => {
    emailFocus.current.focus();
  }, []);

  return (
    <div className='login-content'>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId='formBasicEmail'>
          <Form.Label>Email address</Form.Label>
          <Form.Control
            type='email'
            ref={emailFocus}
            name='email'
            value={user.email}
            placeholder='Enter email'
            onChange={handleChange}
            style={{ width: "300px" }}
          />
          {isError ? <Error error={formValidationErrors.emailError} /> : " "}
        </Form.Group>

        <Form.Group controlId='formBasicPassword'>
          <Form.Label>Password</Form.Label>
          <Form.Control
            type='password'
            name='password'
            value={user.password}
            placeholder='Password'
            onChange={handleChange}
          />
          {isError ? <Error error={formValidationErrors.passwordError} /> : ""}
        </Form.Group>
        <Form.Group controlId='formBasicCheckbox'>
          <Form.Check type='checkbox' label='Check me out' />
        </Form.Group>
        <Button variant='primary' type='submit'>
          Submit
        </Button>
        <p>Didn&apos;t create an account?</p>
        <Link to='/signup'>Signup</Link>
      </Form>
    </div>
  );
}
