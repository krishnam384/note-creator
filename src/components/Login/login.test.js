import React from "react";
import { shallow } from "enzyme";
import Login from "./Login";

/* eslint no-undef: 0 */

describe("Testing Login Comonent", () => {
  it("Snapshot of Login Component", () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Testing Email Input", () => {
    const wrapper = shallow(<Login />);

    wrapper.find("#email").simulate("change", {
      target: { name: "email", value: "sai@gmail.com" },
    });

    expect(wrapper.find("#email").prop("value")).toEqual("sai@gmail.com");
  });
});
