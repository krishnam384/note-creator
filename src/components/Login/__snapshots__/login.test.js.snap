// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login Comonent Snapshot of Login Component 1`] = `
<div
  className="login-content"
>
  <Modal
    footer={
      <button
        onClick={[Function]}
      >
        close
      </button>
    }
    header="Welcome to Modal"
    onCancel={[Function]}
    show={false}
  >
    <p>
      Hello World
    </p>
  </Modal>
  <button
    onClick={[Function]}
  >
    Modal
  </button>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup
      controlId="formBasicEmail"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Email address
      </FormLabel>
      <FormControl
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Enter email"
        style={
          Object {
            "width": "300px",
          }
        }
        type="email"
        value=""
      />
       
    </FormGroup>
    <FormGroup
      controlId="formBasicPassword"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Password
      </FormLabel>
      <FormControl
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </FormGroup>
    <FormGroup
      controlId="formBasicCheckbox"
    >
      <FormCheck
        label="Check me out"
        type="checkbox"
      />
    </FormGroup>
    <Button
      active={false}
      disabled={false}
      type="submit"
      variant="primary"
    >
      Submit
    </Button>
    <p>
      Didn't create an account?
    </p>
    <Link
      to="/signup"
    >
      Signup
    </Link>
  </Form>
</div>
`;
